sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsConnectionStringProperties.cs

```diff
@@ -334,6 +328,17 @@ public static EventHubsConnectionStringProperties Parse(string connectionString)
                         {
                             endpointUri = null;
                         }
+                        else if (string.IsNullOrEmpty(endpointUri.Host) && (CountChar(':', value.AsSpan()) == 1))
```

Comment:
Would it make sense to utilize `Uri.IsAbsoluteUri` and/or `Uri.Port` to calculate this?

Reply:
It gets mangled when you've got something like `localhost:8080` where "localhost" becomes the scheme, the host is empty, and the port sits at the -1 default.    I couldn't figure out a less hacky but reliable way to detect without making a lot more assumptions about the format.

```csharp
    var x = "localhost:1234";
    Uri.TryCreate(x, UriKind.Absolute, out var endpoint).Dump();
    
    endpoint.Scheme.Dump();           // localhost
    endpoint.Host.Dump();                // empty
    endpoint.Port.Dump();                 // -1
    endpoint.AbsoluteUri.Dump();    // equal to the input string
```

```text
True
localhost

-1
localhost:1234
```